terraform/README.md

```diff
@@ -0,0 +1,228 @@
+<!---
+Copyright 2019 Crown Copyright
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+
+
+--->
+
+
+# <img src="logos/logo.svg" width="180">
+
+### EKS terraform deployment
+The terraform here creates an EKS cluster and includes the following resources:
+1. EKS Cluster: AWS managed Kubernetes cluster of master servers.
+1. AutoScaling Group containing 2 m4.large instances based on the latest EKS Amazon Linux 2 AMI: Operator managed Kuberneted worker nodes for running Kubernetes service deployments
+1. Associated VPC, Internet Gateway, Security Groups, and Subnets: Operator managed networking resources for the EKS Cluster and worker node instances
+1. Associated IAM Roles and Policies: Operator managed access resources for EKS and worker node instances
+
+The code has been adapted from the following [link](https://learn.hashicorp.com/terraform/aws/eks-intro)
+
+The code uses a backend configuration (see backend.tf) - the bucket configuration will need to be modified to point to an S3 bucket you create for the storage of state:
+
+Use the following aws command to create a bucket:
+
+```bash
+aws s3 mb s3://<<bucket name to add to backend.tf>> --region eu-west-1
+```
+
+Ensure you have permissions to do the following (this can be done from the aws command line if you as a user have these permissions or alternatively by attaching a role to an EC2 instance with these permissions:
+
+If you are using an EC2 instance for executing the terraform you will need to install terraform v0.12.7 or greater as well as aws-iam-authenticator.
+Create a role as per below and attach this to the EC2 instance. Run the following:
+
+```bash
+curl http://169.254.169.254/latest/meta-data/iam/security-credentials/<<role name>>

```

Comment:
Lets remove the IP address from this line

Reply:
this is a public ip from aws: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html -- see here.